lib/state-service.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+const joi = require('joi');
+const fs = require('fs');
+const mime = require('mime');
+const { featureShema } = require('./routes/admin-api/feature-schema');
+const strategySchema = require('./routes/admin-api/strategy-schema');
+const getLogger = require('./logger');
+const yaml = require('js-yaml');
+const {
+    FEATURE_IMPORT,
+    DROP_FEATURES,
+    STRATEGY_IMPORT,
+    DROP_STRATEGIES,
+} = require('./event-type');
+
+const logger = getLogger('state-service.js');
+
+const dataSchema = joi.object().keys({
+    features: joi
+        .array()
+        .optional()
+        .items(featureShema),
+    strategies: joi
+        .array()
+        .optional()
+        .items(strategySchema),
+});
+
+class StateService {
+    constructor(config) {
+        this.config = config;
+    }
+
+    async importFile({ importFile, dropBeforeImport, userName }) {
+        let data = await new Promise((resolve, reject) =>
```

Comment:
```js
 let data = await new Promise((resolve, reject) =>
            fs.readFile(importFile, (err, v) =>
                err ? reject(err) : resolve(v)
            )
        );
```
can be simplified to:
```js
let data = await fs.readFile(importFile);
```

Reply:
it this possible without fs-extra?

Reply:
no your right. `require('fs').promises` was added in node 10. I don't want to require new node version before next major of unleash.

Reply:
If you are doing the await straight away there is no reason to no just use the sync api for reading the file. 

A async aproach could be:

```js
/// util functions:
function readFile(importFile) {
    return new Promise((resolve, reject) =>
        fs.readFile(importFile, (err, v) => (err ? reject(err) : resolve(v)))
    );
}

function yamlToJs(importFile, data) {
    return mime.lookup(importFile) === 'text/yaml' ? yaml.safeLoad(data) : data;
}
```

using the util functions the importFile could look like:
```js
importFile({ importFile, dropBeforeImport, userName }) {
        return readFile(importFile)
            .then(yamlToJs.bind(this, importFile))
            .then(data => this.import({ data, dropBeforeImport, userName }));
    }

```

Reply:
you don't like await huh 😄 
but i like this approach, done.